### PR TITLE
quote "row_number" keyword (reserved in MySQL 8)

### LINKED ISSUE #3: MySQL8 compatiblity
Test on MYSQL8.

### PR DESCRIPTION
**Pull request by marcing**
fixes #3, #4

### CODE REVIEW

**Comment by falkenhawk** [comment]
thanks, but first we need to setup testing env on travis for both mysql 5.7 & 8, and fix those erroring php 5.4 and 5.5 builds: https://travis-ci.community/t/php-5-4-and-5-5-archives-missing/3723

File: lib/Doctrine/Query.php [hunk 1]
```diff
@@ -1804,7 +1804,7 @@ public function getLimitSubquery()
                 // otherwise limit would be executed after ids are reordered by distinct / group by, and that order cannot be determined.
                 if ($driverName == 'mysql') {
                     // assign an incremented number to each row with id
-                    $subquery = 'SELECT @rownum:=@rownum + 1 as row_number,'
+                    $subquery = 'SELECT @rownum:=@rownum + 1 as `row_number`,'
```
Comment (falkenhawk) [inline lib/Doctrine/Query.php]: This should be quoted properly as follows:
```suggestion
                    $subquery = 'SELECT @rownum:=@rownum + 1 as ' . $this->_conn->quoteIdentifier('row_number') . ','
```
...then the failing test won't be failing anymore.

**Review by falkenhawk — CHANGES_REQUESTED** [review]

**Comment by marcing** [comment]
> 
> 
> thanks, but first we need to setup testing env on travis for both mysql 5.7 & 8, and fix those erroring php 5.4 and 5.5 builds: https://travis-ci.community/t/php-5-4-and-5-5-archives-missing/3723

PHP5 is EOL. It's a waste of time....

**Comment by milanorszagh** [comment]
I don't feel certain enough in this to review it, but I will happily share my thoughts on it.

**Comment by falkenhawk** [comment]
>  first we need to setup testing env on travis for both mysql 5.7 & 8

actually the test suite never really connects to the database, it only mocks connections. Few tests which used a real connection have been long commented out in the original repo. I am afraid it's not feasible to add mysql 5.7/8 specific tests with real connection in a reasonable time. However, they might be necessary in the future.

Since this change has been tested for some time on a project, and no other issues with mysql 8 popped up, I could optimistically assume that this doctrine1 fork is compatible with mysql 8...

**Review by falkenhawk — APPROVED** [review]

**Pull request merged**